src/connector/src/source/iceberg/parquet_file_handler.rs

```diff
@@ -109,16 +109,23 @@ pub fn new_s3_operator(
     s3_access_key: String,
     s3_secret_key: String,
     bucket: String,
+    is_minio: bool,
 ) -> ConnectorResult<Operator> {
     // Create s3 builder.
-    let mut builder = S3::default().bucket(&bucket).region(&s3_region);
-    builder = builder.secret_access_key(&s3_access_key);
-    builder = builder.secret_access_key(&s3_secret_key);
-    builder = builder.endpoint(&format!(
-        "https://{}.s3.{}.amazonaws.com",
-        bucket, s3_region
-    ));
-
+    let mut builder = S3::default();
+    builder = builder
+        .region(&s3_region)
+        .access_key_id(&s3_access_key)
+        .secret_access_key(&s3_secret_key)
+        .bucket(&bucket);
+    builder = match is_minio {
+        true => builder.endpoint(&format!("http://{}.127.0.0.1:9301", bucket)),
```

Comment:
Should we allow users to provide s3 endpoint instead of hardcoding the endpoint?

Reply:
I think minio is only for test, so I hardcode it. IMO, we don't public `file_scan(minio)` to user, what do you think?

Reply:
>Should we allow users to provide s3 endpoint instead of hardcoding the endpoint?

And this will require another parameter in file scan, which is a bit redudant.

Reply:
Can we reuse the s3 region parameter position to allow users to set the endpoint as well, i.e. (region or endpoint)?

Reply:
region is a required field. If you change region to endpoint, the user's endpoint must include region. Is this a bit strange?

Reply:
When testing minio, the region is `custom`, the endpoint is `http://bucket/127.0.0.1:9301`, they have no intersection.

Reply:
If an endpoint is specified, the region field is useless (Correct?).  I mean `file_scan('parquet', 's3', region_or_endpoint, ...)` the `region_or_endpoint` parameter could be region or endpoint, we can detect it simply by `http` prefix.

Reply:
Let me take a try.